lua/xcodebuild/platform/macos.lua

```diff
@@ -11,21 +11,28 @@ local M = {}
 
 ---Simply starts the application on macOS.
 ---@param appPath string
+---@param productName string
 ---@param callback function|nil
 ---@return number # job id
-function M.launch_app(appPath, callback)
-  local command = { "open", appPath }
+function M.launch_app(appPath, productName, callback)
+  local executablePath = appPath .. "/Contents/MacOS/" .. productName
+  local args = table.concat(appdata.read_run_args() or {}, " ")
+  local command = executablePath .. " " .. args
```

Comment:
We need to keep `command` as a table. Concatenation won't work well here because if the argument contains a space, it will be treated as two parameters.